lib/mayaUsd/fileio/utils/meshWriteUtils.cpp

```diff
@@ -377,4 +405,76 @@ UsdMayaMeshUtil::exportReferenceMesh(UsdGeomMesh& primSchema, MObject obj)
     primVar.GetAttr().Set(VtValue(points));
 }
 
+void
+UsdMayaMeshUtil::assignSubDivTagsToUSDPrim(MFnMesh& meshFn,
+                                           UsdGeomMesh& primSchema,
+                                           UsdUtilsSparseValueWriter& valueWriter)
+{
+    // Vert Creasing
+    MUintArray mayaCreaseVertIds;
+    MDoubleArray mayaCreaseVertValues;
+    meshFn.getCreaseVertices(mayaCreaseVertIds, mayaCreaseVertValues);
+    if (!TF_VERIFY(mayaCreaseVertIds.length() == mayaCreaseVertValues.length())) {
+        return;
+    }
+    if (mayaCreaseVertIds.length() > 0u) {
+        VtIntArray subdCornerIndices(mayaCreaseVertIds.length());
+        VtFloatArray subdCornerSharpnesses(mayaCreaseVertIds.length());
+        for (unsigned int i = 0u; i < mayaCreaseVertIds.length(); ++i) {
+            subdCornerIndices[i] = mayaCreaseVertIds[i];
+            subdCornerSharpnesses[i] = mayaCreaseVertValues[i];
+        }
+
+        // not animatable
+        UsdMayaWriteUtil::SetAttribute(primSchema.GetCornerIndicesAttr(), &subdCornerIndices, valueWriter);
+
+        // not animatable
+        UsdMayaWriteUtil::SetAttribute(primSchema.GetCornerSharpnessesAttr(), &subdCornerSharpnesses, valueWriter);
+    }
+
+    // Edge Creasing
+    int edgeVerts[2];
+    MUintArray mayaCreaseEdgeIds;
+    MDoubleArray mayaCreaseEdgeValues;
+    meshFn.getCreaseEdges(mayaCreaseEdgeIds, mayaCreaseEdgeValues);
+    if (!TF_VERIFY(mayaCreaseEdgeIds.length() == mayaCreaseEdgeValues.length())) {
+        return;
+    }
+    if (mayaCreaseEdgeIds.length() > 0u) {
+        std::vector<int> subdCreaseIndices(mayaCreaseEdgeIds.length() * 2);
+        std::vector<float> subdCreaseSharpnesses(mayaCreaseEdgeIds.length());
+        for (unsigned int i = 0u; i < mayaCreaseEdgeIds.length(); ++i) {
+            meshFn.getEdgeVertices(mayaCreaseEdgeIds[i], edgeVerts);
+            subdCreaseIndices[i * 2] = edgeVerts[0];
+            subdCreaseIndices[i * 2 + 1] = edgeVerts[1];
+            subdCreaseSharpnesses[i] = mayaCreaseEdgeValues[i];
+        }
+
+        std::vector<int> numCreases;
+        std::vector<int> creases;
+        std::vector<float> creaseSharpnesses;
+        compressCreases(subdCreaseIndices, subdCreaseSharpnesses,
+                &numCreases, &creases, &creaseSharpnesses);
```

Comment:
I assume compressCreases is inline? (or an anon method without any DSO export linkeage). 

If this results in an actual function call, none of the arrays you're passing in will ever be able to vectorise. If it's inline, you'll be good.

Reply:
@robthebloke I am not sure if I understand the connection between inline and vectorization?

From what I know, inline has nothing to do with vectorization and I quote:

https://en.cppreference.com/w/cpp/language/inline
> The original intent of the inline keyword was to serve as an indicator to the optimizer that inline substitution of a function is preferred over function call, that is, instead of executing the function call CPU instruction to transfer control to the function body, a copy of the function body is executed without generating the call. This avoids overhead created by the function call (passing the arguments and retrieving the result) but it may result in a larger executable as the code for the function has to be repeated multiple times.
> Since this meaning of the keyword inline is non-binding, compilers are free to use inline substitution for any function that's not marked inline, and are free to generate function calls to any function marked inline. Those optimization choices do not change the rules regarding multiple definitions and shared statics listed above.